From: Vikas Chaudhary <vikas.chaudhary@qlogic.com>
Date: Thu, 13 Feb 2014 10:04:23 +0530
Subject: [PATCH 31/34] qla4xxx: Initialize hardware queue for ISP40XX
Git-commit: aa6db9b9cdf642b220f2a0861facf6bab1d3ab37
Patch-mainline: v3.16-rc1
References: FATE#317545 bnc#909371

Issue:
System crash while target discovery for ISP40XX

Root cause:
Function qla4xxx_init_rings() is not called for ISP40XX

Fix:
Call function qla4xxx_init_rings() for ISP40XX from
qla4xxx_start_firmware().

Signed-off-by: Vikas Chaudhary <vikas.chaudhary@qlogic.com>
Reviewed-by: Mike Christie <michaelc@cs.wisc.edu>
Signed-off-by: Christoph Hellwig <hch@lst.de>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/scsi/qla4xxx/ql4_init.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/scsi/qla4xxx/ql4_init.c b/drivers/scsi/qla4xxx/ql4_init.c
index 1efea1d..b36e3f4 100644
--- a/drivers/scsi/qla4xxx/ql4_init.c
+++ b/drivers/scsi/qla4xxx/ql4_init.c
@@ -864,6 +864,8 @@ int qla4xxx_start_firmware(struct scsi_qla_host *ha)
 	if (status == QLA_SUCCESS) {
 		if (test_and_clear_bit(AF_GET_CRASH_RECORD, &ha->flags))
 			qla4xxx_get_crash_record(ha);
+
+		qla4xxx_init_rings(ha);
 	} else {
 		DEBUG(printk("scsi%ld: %s: Firmware has NOT started\n",
 			     ha->host_no, __func__));
-- 
1.8.0

